src/main.rs

```diff
@@ -177,188 +225,225 @@ async fn sync(
+            .map(|link| link.url.clone());
+
+        if let Some(next_url) = next_url {
+            url = next_url;
+        } else {
+            break;
         }
     }
     //insert or update the last remaining resources
     if update_batch.len() > 0 {
-        update_helper(pg_client, &update_batch, table_name).await?;
+        update_counter += update_batch.len() as u32;
+        update_helper(pg_con_pool, &update_batch, table_name).await?;
     }
     if insert_batch.len() > 0 {
-        insert_helper(pg_client, &insert_batch, table_name).await?;       
+        insert_counter += insert_batch.len() as u32;
+        insert_helper(pg_con_pool, &insert_batch, table_name).await?;       
     }
     //remove rows from pg that were not encountered in blaze
     let delete_ids: Vec<i32> = pg_versions.values().map(|value| value.pk_id).collect();
-    if delete_ids.len() > 0 {
-        delete_helper(pg_client, &delete_ids, table_name).await?;
+    if delete_ids.len() > 0 {        
+        delete_helper(pg_con_pool, &delete_ids, table_name).await?;
     }
+
+    info!("Updated {} rows", update_counter);
+    info!("Inserted {} rows", insert_counter);
+    info!("Deleted {} rows", delete_ids.len() as u32);
+
+    //compare total entry counts between blaze and pg
+    let row_count = get_num_rows(pg_con_pool, table_name).await?;
+    let resource_count = get_blaze_resource_count(blaze_base_url, type_arg).await?;
+    if row_count != resource_count {
+        warn!("{} entry counts do not match between Blaze and PostgreSQL", type_arg);
+    } else {
+        info!("{} entry counts match between Blaze and PostgreSQL \u{2705}", type_arg);
+    }
+
+    Ok(())
+}
+
+
+pub async fn run_sync(pg_con_pool: &PgPool, config: &Config) -> Result<(), anyhow::Error>{
+    let type_args: Vec<&str> = vec!["Specimen", "Patient", "Observation", "Condition"];
+
+    let table_names: Vec<&str> = vec!["specimen", "patients", "observations", "conditions"];
     
+    //check preconditions for sync    
+    let blaze_available = check_blaze_connection(&config.blaze_base_url,
+         config.blaze_num_connection_attempts).await?;
+    let all_tables_exist = pred_tables_exist(pg_con_pool, &table_names).await?;
+
+    if blaze_available && all_tables_exist {
+        info!("All tables found as expected");
+        for type_arg in type_args {
+            sync_blaze_2_pg(
+                &config.blaze_base_url,
+                pg_con_pool, 
+                type_arg,
+                config.pg_batch_size, 
+                config.blaze_page_resource_count).await?;
+        }
+    } else if blaze_available && !all_tables_exist {
+        create_tables(pg_con_pool).await?;
+        for type_arg in type_args {
+            sync_blaze_2_pg(
+                &config.blaze_base_url,
+                pg_con_pool,
+                type_arg,
+                config.pg_batch_size,
+                config.blaze_page_resource_count).await?;
+        }
+    }
     Ok(())
```

Comment:
I find the control flow of this function a bit weird. The function will return `Ok` if blaze is not available?